File: backend/src/main/java/com/campuspilot/entity/Task.java
package com.campuspilot.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@TableName("tasks")
public class Task {
    
    @TableId(type = IdType.AUTO)
    private Long id;
    
    private Long userId;
    
    private Long policyId;
    
    private String title;
    
    private String description;
    
    private String status;
    
    private String priority;
    
    private LocalDateTime deadline;
    
    private LocalDateTime completedAt;
    
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;
    
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;
    
    private Long createdBy;
    
    private Long updatedBy;
    
    @TableLogic
    private Integer isDeleted;
}

File: campuspilot-web/src/api/task.ts
import api from './auth'

export interface Task {
  id: number
  title: string
  description: string
  status: string
  priority: string
  deadline: string
  createdAt: string
  updatedAt: string
}

export function getTasks() {
  return api.get('/tasks')
}

export function getTaskById(id: number) {
  return api.get(`/tasks/${id}`)
}

export function createTask(task: { title: string; description?: string; deadline?: string }) {
  return api.post('/tasks', task)
}

export function completeTask(id: number) {
  return api.put(`/tasks/${id}/complete`)
}

export function cancelTask(id: number) {
  return api.put(`/tasks/${id}/cancel`)
}